### PR TITLE
migrate Alignment/TrackerAlignment to use esConsumes

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The purpose of this PR is to modernize the `Alignment/TrackerAlignment` subsystem, points addressed are:
   * `esConsumes` migration: part of #31061
   * correct one mistake accidentally introduced in https://github.com/cms-sw/cmssw/pull/32005 

#### PR validation:

Branch compiles and run the unit tests successfully.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

This is not a backport and no backport is needed.

### CAVEAT 
There is a known potential merge conflict with PR  https://github.com/cms-sw/cmssw/pull/32278, once that's merged I will rebase mine.

### CODE REVIEW

File: Alignment/TrackerAlignment/plugins/MisalignedTrackerESProducer.cc [hunk 1]
```diff
@@ -59,33 +63,31 @@ class MisalignedTrackerESProducer : public edm::ESProducer {
 
 //__________________________________________________________________________________________________
 MisalignedTrackerESProducer::MisalignedTrackerESProducer(const edm::ParameterSet& p)
-    : theSaveToDB(p.getUntrackedParameter<bool>("saveToDbase")),
+    : geomDetToken_(setWhatProduced(this).consumes()),
```
Comment (mmusich) [inline Alignment/TrackerAlignment/plugins/MisalignedTrackerESProducer.cc:66]: @makortel for my own education, is this syntax acceptable also when there are multiple tokens, or shall I instead use a different one:

```c++
auto cc = setWhatProduced(this);
geomDetToken_ = cc.consumes();
ptpToken_ = cc.consumes();
topoToken_ = cc.consumes();
```
in the body of the constructor?

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/TrackerAlignment/plugins/MisalignedTrackerESProducer.cc (hunk 1 above)
Comment (makortel) [reply Alignment/TrackerAlignment/plugins/MisalignedTrackerESProducer.cc:66]: The `setWhatProduced(this)` should be called exactly once, so you need to use the other pattern of

> ```c++
> auto cc = setWhatProduced(this);
> geomDetToken_ = cc.consumes();
> ptpToken_ = cc.consumes();
> topoToken_ = cc.consumes();
> ```
> 
> in the body of the constructor

Comment (mmusich) [reply Alignment/TrackerAlignment/plugins/MisalignedTrackerESProducer.cc:66]: thanks, will change accordingly.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by christopheralanwest** [comment]
+1

- `esConsumes` migration completed for this package
- no unexpected changes in PR comparison tests

**Comment by silviodonato** [comment]
+1

**Pull request merged**